src/devices/src/virtio/net/device.rs

```diff
@@ -98,6 +99,7 @@ impl Net {
         guest_mac: Option<&MacAddr>,
         rx_rate_limiter: RateLimiter,
         tx_rate_limiter: RateLimiter,
+        mmds_ipv4_addr: Option<Ipv4Addr>,
         allow_mmds_requests: bool,
```

Comment:
`mmds_ipv4_addr` doesn't make sense when `allow_mmds_requests == false`.

The two args can be combined in a single enum, smth like:
```rust
enum MmdsConfig {
    Disabled,
    EnabledDefaultIp,
    EnabledCustomIp(Ipv4Addr),
}
```

Reply:
Another option would be to just pass Option<MmdsNetworkStack> and create that during the building phase instead of passing configuration and having the network device build it.

Reply:
As a matter of fact, this issue describes on the same lines but shorter what I have in mind: https://github.com/firecracker-microvm/firecracker/issues/1761.